lib/OpenQA/Task/Job/HookScript.pm

```diff
@@ -1,28 +1,40 @@
-# Copyright 2022 SUSE LLC
+# Copyright 2022 SUSE LLC3
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 package OpenQA::Task::Job::HookScript;
 use Mojo::Base 'Mojolicious::Plugin', -signatures;
 use OpenQA::Task::SignalGuard;
+use Data::Dumper;
+$Data::Dumper::Indent = 1;
+$Data::Dumper::Sortkeys = 1;
+
+use constant LINEAR_BACKOFF_BASE_SECONDS => 40;
```

Comment:
So currently it's possible to influence the delay via an environment variable, but you want to make it a hardcoded constant now, and the delay passed via the `$options` is only used when it's 0?

My suggestion would be: The delay of one minute is ok for the initial retry, and we can simply use the existing `$options->{delay}`, and then just multiply it with the `$job->retries` as you have already done.

Reply:
Currently there is a delay of 1 minute, and it is retrying 1440 times, waiting 1 minute in between.
One hook script run takes around 10 seconds, meaning that 6 hook script jobs all retrying every minute make a whole minion worker slot busy (10 seconds * 6 jobs retrying every minute).

By retrying less often, e.g. increasing the delay linearly, we make the minion server less busy, so it can pick up jobs that would otherwise pile up.

The initial delay of 1 minute is totally fine.

Reply:
And if you wanted to change that initial delay, you could do so by setting the environment variable or job setting, so I don't know why you want to use a hard coded constant now instead.